File: WOT-iOS/WOT-iOS/core/requests/WOTWebRequest/Radios/WOTWebRequestTankRadios.m
//
//  WOTWebRequestTankRadios.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/23/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTWebRequestTankRadios.h"

@implementation WOTWebRequestTankRadios

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOT_KEY_APPLICATION_ID] = [NSString valueOrSpaceString:self.applicationID];
    result[WOT_KEY_FIELDS] = [NSString valueOrSpaceString:self.args[WOT_KEY_FIELDS]];
    result[WOT_KEY_MODULE_ID] = [NSString valueOrSpaceString:self.args[WOT_KEY_MODULE_ID]];
    return result;
}

- (NSString *)path {
    
    return @"wot/encyclopedia/tankradios/";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankRadios.swift
//
//  WOTWebRequestTankRadios.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankRadios: WOTWEBRequest, WOTModelServiceProtocol {

    override public var method: String { return "POST" }
    
    @available(*, deprecated, message: "TO be refactored")
    @objc
    public class func modelClassName() -> String {
        return NSStringFromClass(Tankradios.self)
    }

    @available(*, deprecated, message: "TO be refactored")
    @objc
    public func instanceModelClass() -> AnyClass? {
        return NSClassFromString( type(of: self).modelClassName() )
    }

    override public var path: String {
        return "/wot/encyclopedia/tankradios/"
    }
}
